### PR TITLE
Adding proxy support to HTTP

### PR DESCRIPTION
**Pull request by mconnew**
Addresses #1592

### CODE REVIEW

File: src/System.Private.ServiceModel/src/System.Private.ServiceModel.csproj [hunk 1]
```diff
@@ -70,4 +70,4 @@
     </Content>
   </ItemGroup>
   <Import Project="$([MSBuild]::GetDirectoryNameOfFileAbove($(MSBuildThisFileDirectory), dir.targets))\dir.targets" />
-</Project>
+</Project>
```
Comment (zhenlan) [inline src/System.Private.ServiceModel/src/System.Private.ServiceModel.csproj]: NIT: This file is not really changed.

File: src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceModelHttpMessageHandler.Window.cs [hunk 2]
```diff
@@ -0,0 +1,95 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
```
Comment (zhenlan) [inline src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceModelHttpMessageHandler.Window.cs]: Should we name this file *.CoreCLR.Windows.cs?

File: src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceModelHttpMessageHandler.CoreClr.cs [hunk 3]
```diff
@@ -10,7 +10,7 @@
 
 namespace System.ServiceModel.Channels
```
Comment (zhenlan) [inline src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceModelHttpMessageHandler.CoreClr.cs]: Wondering if we should rename this file *.CoreClr.Unix.cs?

Comment (mconnew) [inline src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceModelHttpMessageHandler.CoreClr.cs]: Done

File: src/System.Private.ServiceModel/src/System.Private.ServiceModel.csproj (hunk 1 above)
Comment (mconnew) [inline src/System.Private.ServiceModel/src/System.Private.ServiceModel.csproj]: Fixed

File: src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceModelHttpMessageHandler.Window.cs (hunk 2 above)
Comment (mconnew) [inline src/System.Private.ServiceModel/src/System/ServiceModel/Channels/ServiceModelHttpMessageHandler.Window.cs]: Done.

**Comment by mconnew** [comment]
@dotnet-bot test all outerloop please

**Comment by mconnew** [comment]
@dotnet-bot test outerloop Ubuntu14.04 Release
@dotnet-bot test outerloop OpenSUSE13.2
@dotnet-bot test outerloop Fedora23

**Comment by mconnew** [comment]
@dotnet-bot test Innerloop Windows_NT Debug

**Comment by mconnew** [comment]
Merging even though some checks failed as we have never run on Fedora23 or OpenSUSE13.2 and those runs are an artifact or something happening with the CI system unrelated to this PR.

**Pull request merged**